cluster/store/db.go

```diff
@@ -29,10 +30,11 @@ var (
 )
 
 type localDB struct {
-	Schema *schema
-	store  Indexer
-	parser Parser
-	log    *slog.Logger
+	schemaLock sync.Mutex
```

Comment:
The Mutex might not be needed at this point, since schema in schema.go handles concurrent access. 
Please see my comment below (store.go) for more details